### PR TITLE
Detailed user information not requested if the users closes the app at registration

### PR DESCRIPTION
**Pull request by rishabhkohli**

### CODE REVIEW

File: app/src/main/java/app/nexd/android/ui/MainActivity.kt [hunk 1]
```diff
@@ -26,8 +26,13 @@ class MainActivity : AppCompatActivity() {
...
                     Preferences.getToken(this)?.let {
-                        Log.v("Navigation", "redirect to roleFragment")
-                        controller.navigate(R.id.roleFragment)
+                        if (Preferences.getRegistrationComplete(this)) {
+                            Log.v("Navigation", "redirect to roleFragment")
+                            controller.navigate(R.id.roleFragment)
+                        } else {
+                            Log.v("Navigation", "redirect to registerDetailedViewModel")
+                            controller.navigate(R.id.registerDetailedFragment)
+                        }
```
Comment (andre-stefanov) [inline app/src/main/java/app/nexd/android/ui/MainActivity.kt:35]: this logic should happen in a viewmodel

Comment (rishabhkohli) [reply app/src/main/java/app/nexd/android/ui/MainActivity.kt]: Done.

**Comment by andre-stefanov** [comment]
Please resolve merge conflicts

**Comment by rishabhkohli** [comment]
Rebased the branch since the changes to develop were very substantial.
Also squashed the earlier 2 commits into one, because the earlier structure did not make any sense anymore.
Lastly, there was a bug on develop involving a crash on the RegisterDetailedFragment.
Fixed that in a separate commit on this branch.

**Pull request merged**